MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,446 @@
+    NSDictionary *attributes = @{
+                                 @"id":@"_id",
+                                 @"monster_class":@"class",
+                                 @"name":@"name",
+                                 @"jpn_name":@"jpn_name",
+                                 @"sort_name":@"sort_name",
+                                 @"trait":@"trait",
+                                 @"icon":@"icon_name",
+                                 @"url":@"url"
+                                 };
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:[Monster entityName]];
+    path = [[NSBundle mainBundle] pathForResource:MonsterDamageZonesFileName ofType:JSON];
+    attributes = @{
+                   @"id":@"_id",
+                   @"monster_id":@"monster_id",
+                   @"monsterName":@"monster_name",
+                   @"jpnMonsterName":@"jpn_monster_name",
+                   @"bodyPart":@"body_part",
+                   @"cut":@"cut",
+                   @"impact":@"impact",
+                   @"shot":@"shot",
+                   @"fire":@"fire",
+                   @"water":@"water",
+                   @"thunder":@"thunder",
+                   @"ice":@"ice",
+                   @"dragon":@"dragon",
+                   @"ko":@"ko",
+                   @"extract":@"extract"
+                   };
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:[DamageZone entityName]];
+}
+
+- (void)loadMonsterDropData
+{
+    NSArray *dropList = [self loadArrayFromJsonFileNamed:MonsterDropsFileName];
+    for (NSDictionary *drop in dropList) {
+        // Only add the drop if it is not already there
+        NSNumber *monsterDropID = drop[MonsterDropIDKey];
+        NSPredicate *monsterDropPredicate = [NSPredicate predicateWithFormat:@"%K == %d", MonsterDropIDKey, monsterDropID.intValue];
+        if (![self countForEntityWithEntityName:[MonsterDrop entityName] withPredicate:monsterDropPredicate]) {
+            MonsterDrop *newDrop = [MonsterDrop insertInManagedObjectContext:self.managedObjectContext];
+            newDrop.id = drop[MonsterDropIDKey];
+            newDrop.method = drop[MonsterDropMethodKey];
+            newDrop.quantity = drop[MonsterDropQuantityKey];
+            
+            // If the Item for this drop already exists in the persistent store ...
+            NSPredicate *itemPredicate = [NSPredicate predicateWithFormat:@"%K == %@", ItemNameKey, drop[MonsterDropItemNameKey]];
+            Item *item = (Item *)[self getUniqueEntityWithEntityName:[Item entityName] withPredicate:itemPredicate];
+            if (item) {
+                // We need to hook up relationships appropriately.
+                newDrop.item = item;
+            } else {
+                // Otherwise, we should create the item.
+                Item *newItem = [Item insertInManagedObjectContext:self.managedObjectContext];
```

Comment:
If you used the `item` variable that already exists, you could move the assignment that's basically duplicated as the last line of both the `if` and `else` blocks outside the conditional.